### PR TITLE
feat: `bot.isRunning()`

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Adds a way to inspect whether or not the bot is currently running.

### CODE REVIEW

File: src/bot.ts [hunk 1]
```diff
@@ -495,6 +498,25 @@ a known bot info object.",
...
+     * and false otherwise.
+     *
+     * If this method returns true, it means that `bot.start()` has been called,
+     * and that the bot has neither crashed nor was it stopped via a call to
+     * `bot.stop()`. This also means that you cannot use this method to check if
+     * a webhook server is running, or if grammY runner was started.
+     *
+     * Note that this method will already begin to return true even before the
+     * call to `bot.start()` has completed its initialization phase. During that
+     * period, it is `bot.isRunning() && !bot.isInited()`. By extension, this
```
Comment (roziscoding) [inline src/bot.ts:512]: > During that period, it is `bot.isRunning() && !bot.isInited()`

I don't get what this is trying to tell me. Does it mean the function returns something else? Or that I should do something else. I think could be a bit more clear here 🤔

Comment (KnorpelSenf) [reply src/bot.ts:512]: It just means that just because the bot is running, does not mean it is inited. I will clarify this.

**Review by KnightNiwrem — APPROVED** [review]

**Comment by KnightNiwrem** [comment]
Interesting... so this never worked before:

```ts
// Some transformer to always throw 401 error on getUpdates

try {
  await bot.start();
} catch {}

console.log('abc');
console.log(bot.pollingRunning);

try {
  await bot.start();
} catch {}
```

```txt
grammy:bot Initializing bot +0ms
grammy:core Calling getMe +0ms
grammy:core Calling deleteWebhook +9ms
grammy:bot I am he_exp_bot! +605ms
grammy:bot Starting simple long polling +2ms
grammy:error msg (401: abc) +0ms
abc
true
grammy:bot Simple long polling already running! +3ms
```

**Comment by KnorpelSenf** [comment]
Correct. That was a bug.

**Pull request merged**